File: star_seller.vb
'vim: ft=vb
Import System

''' The Star Seller exercise in algorithmics,
''' written in Visual Basic .NET.           [22 lines of code]

''' This is the very first programming language I got to try. I used it on
''' Future Pinball, then started using Visual Basic 2010 Express.
''' Luckily, I stopped using it as I learned more about programming in general.
''' The syntax for this language is kind of heavy in retrospect, though I
''' remember having noted that before.

Module StarSeller
    Sub Main(args as String())
        Dim Name    as String
        Dim Stars   as Integer
        Dim Counter as Integer
        Dim Valid   as Boolean = False

        Console.Write("What is your name? ")
        Name = Console.ReadLine()
        Console.WriteLine("Hello, " + Name + "!")
        While Not Valid
            Console.Write("How many stars do you want? ")
            Stars = Console.ReadLine()
            If Stars >= 0 And Stars <= 200 Then
                Valid = True
            End If
        End While
        For Counter = 1 To Stars
            Console.Write("*")
        Next
        Console.WriteLine(vbCrLf + "Goodbye, " + Name)
    End Sub
End Module

''' This language is case-insensitive.

File: star_seller.pas
(* The Star Seller exercise in algorithmics,
   written in FreePascal.                   [25 lines of code]
 *)

{ Pascal was likely the prime competitor for C back in the nineties, and their
   fates are surprisingly similar -- they both had an object-oriented evolution
   and they were the first-class language on respective operating systems
   -- Pascal was to Mac OS Classic what C is to UNIX.

  Also: Borland's Delphi is just an evolution of Apple's Object Pascal.
}

var Stars : integer;
    Name  : array[1..20] of char;
    valid : boolean;
    i     : integer;

begin
    valid := False;
    WHILE NOT valid DO
    begin
        write('What is your name? '); readln(Name);
        IF NOT (name = '') THEN valid := True;
    end;
    valid := False;
    writeln('Hello, ', Name, '!');
    WHILE NOT valid DO
    begin
        write('How many stars do you want? '); readln(Stars);
        IF (Stars >= 0) AND (Stars <= 200) THEN
        begin
            FOR i := 1 TO Stars DO write('*');
            writeln;
            valid := True;
        end;
    end;
    writeln('Goodbye, ', Name);
end.

{ This language is case-insensitive. }
